### ISSUE TITLE
WireMockServer.port() does not match environment property

### ISSUE THREAD

**Issue by bertramn** [spring-cloud/spring-cloud-contract#1247]
Bug report

Spring Cloud version 2.1.3.RELEASE

I was hoping to use the `${wiremock.server.port}` property in an integration test to wire up some other component but found that the port number property does not always match the actual port the WireMockServer listens on.

I used the property in an `application-test.yml` but the port is different once this configuration is loaded by a spring boot app.

```yaml
---
resources:
  - id: mock-endpoint-a
    url: "http://localhost:${wiremock.server.port}/mock-call"
```

This simple spring boot unit test also does not work:

```java
import com.github.tomakehurst.wiremock.WireMockServer;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static org.junit.Assert.assertEquals;

@RunWith(SpringRunner.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@AutoConfigureWireMock(port = 0)
@ActiveProfiles("test")
public class WireMockServerPortTest {

  @Autowired
  private WireMockServer wireMockServer;

  @Value("${wiremock.server.port}")
  private int wireMockServerPortProperty;

  // FAILS
  @Test
  public void itShouldMAtchEnvironmentProperty() {
    assertEquals(wireMockServer.port(), wireMockServerPortProperty);
  }

}
```

**Comment by bertramn** [comment]
To make matters worse, hardwiring the port using `@AutoConfigureWireMock(port = 10666)` does not work either. The WireMockServer injected into the test case still listens on a different port than configured in the annotation.

**Comment by marcingrzejszczak** [comment]
Also I wonder if you have this problem when you don't have an active profile

**Comment by bertramn** [comment]
Hmm I have been trying to make this a super simple scenario but for some reason as soon as I start adding a `WireMockConfigurationCustomizer` to the test configuration, I get the following error:

`ApplicationContextException: Unable to start ServletWebServerApplicationContext due to missing ServletWebServerFactory bean.`

https://github.com/bertramn/webmvc-springboot-demo/tree/feature/wiremock-port-issue

I'll keep on trying to replicate the issue when I get a spare minute. For now I ran out of time and dropped the idea of using spring cloud wiremock as the configuration we have to customize has to happen before the spring security configuration is loaded. I also noticed that once we had the internal config that the wiremock configuration was executed multiple times and the WireMockServer singleton bean was registered again despite already being registered.
